test/shmemx/c11_test_shmemx_test_vector.c

```diff
@@ -0,0 +1,180 @@
+#include <shmem.h>
+#include <shmemx.h>
+#include <stdlib.h>
+#include <stdio.h>
+
+#if defined(__STDC_VERSION__) && __STDC_VERSION__ >= 201112L
+
+#define TEST_SHMEM_TEST_ALL(TYPE)                                                                                    \
+    do {                                                                                                             \
+                                                                                                                     \
+        const int my_pe = shmem_my_pe();                                                                             \
+        const int npes = shmem_n_pes();                                                                              \
+        TYPE *ivars = shmem_calloc(npes, sizeof(TYPE));                                                              \
+        int *status = calloc(npes, sizeof(int));                                                                     \
+        TYPE *cmp_values = malloc(npes *sizeof(TYPE));                                                               \
+                                                                                                                     \
+        size_t i = 0;                                                                                                \
+        int expected_sum = 0;                                                                                        \
+        int total_sum = 0;                                                                                           \
+                                                                                                                     \
+        for(i = 0; i<npes; i++){                                                                                     \
+            shmem_p(&ivars[my_pe], (TYPE)my_pe, i);                                                                  \
```

Comment:
This actually needs to be `shmem_atomic_set` instead of `shmem_p`...  We recently changed the wait/test routines to only be compatible with the AMOs.  If interested, you can read about it on PR https://github.com/openshmem-org/specification/pull/267 and issue https://github.com/openshmem-org/specification/issues/248 (the issue might be a little confusing since there's a bit of back-and-forth on the semantics).

Reply:
I actually had shmem_atomic_set here earlier. But I realized that it doesn't support the datatype short so for that reason I used shmem_p. Is it still the case with shmem_atomic _set or has it changed? If it's changed then I can replace this with shmem_atomic_set. But then we won't be able to have a test case for short datatype.

Reply:
Great catch!  And thanks for posting the issue to the spec (https://github.com/openshmem-org/specification/issues/334).

I would say for now we can remove `short` and `unsigned short` from this test...